internal/openfga/interfaces.go

```diff
@@ -30,3 +30,12 @@ type OpenFGAClientInterface interface {
 	ListObjects(context.Context) client.SdkClientListObjectsRequestInterface
 	ListObjectsExecute(client.SdkClientListObjectsRequestInterface) (*client.ClientListObjectsResponse, error)
 }
+
+// OpenFGAClientInterface is the interface used to decouple the OpenFGA store implementation
+type OpenFGAClientInterface interface {
+	ListObjects(context.Context, string, string, string) ([]string, error)
+	ReadTuples(context.Context, string, string, string, string) (*client.ClientReadResponse, error)
+	WriteTuples(context.Context, ...Tuple) error
+	DeleteTuples(context.Context, ...Tuple) error
+	Check(context.Context, string, string, string, ...Tuple) (bool, error)
```

Comment:
question: is this intended as a "Find" store operation? Or should we put it under the Core interface since it's a check on the tuples?

I'm good either way

Reply:
interface is only meant to be used by https://github.com/canonical/identity-platform-admin-ui/pull/382/files#diff-0c2915818e34863217890b989cf6bcaf87f9daaab08e908a2825257a3aab38e8R29

and that is basically our client wrapper around the `core` openfga client https://github.com/canonical/identity-platform-admin-ui/pull/382/files#diff-5aa2f2c8ad62191825fc8a94d7a9e265076d95f62428b8e41f6012e6974035c1R12

now naming convention is terrible here, as if someone decides to use this interface it will get confused


we cannot put those 2 checks in the same interface as this is the "user friendly" one, the `find` method should be implemented on the store but it was not needed atm

plan is to add it when we refactor roles and groups